akka-stream/src/main/scala/akka/stream/javadsl/Graph.scala

```diff
@@ -401,6 +404,26 @@ object GraphDSL extends GraphCreate {
    */
   def builder[M](): Builder[M] = new Builder()(new scaladsl.GraphDSL.Builder[M])
 
+  /**
+   * Creates a new [[Graph]] by importing the given graph list `graphs` and passing their [[Shape]]s
+   * along with the [[GraphDSL.Builder]] to the given create function.
+   */
+
+  def create[IS <: Shape, S <: Shape, M, G <: Graph[IS, M]](
+    graphs: java.util.List[G],
+    block:  function.Function2[GraphDSL.Builder[Seq[M]], java.util.List[IS], S]): Graph[S, java.util.List[M]] = {
+    def transformToJava(x: Seq[M]) = x.asJava
+    val g = scaladsl.GraphDSL.create(graphs.asScala.toList) { b ⇒ seq ⇒ block(b.asJava, seq.asJava) }
```

Comment:
So the problem of `GraphDSL.Builder[Seq[M]]` in the java api come from the builder, because I can't transform the type inside the builder (for ex. I can't map it or at least I didn't find a suitable way to do so). Other option would be to duplicate the create code (from the scala api) and reimplement the functionality with pure java outer interfaces, but I think that Seq is only an inconvenient visibility thing and the actual java user will never need to interact with it. So I see why it is not good, but I can't change it (without duplicating logic), and not really think it's need to be changed :D

Reply:
It matters if you want to use the materialized values inside of the GraphDSL to emit them into the graph, like so: `Outlet<M> mats = builder.materializedValue()`. Looking at the Scala version it's just 6 lines of code, I think it would be worth duplicating those.